src/dependency_downloader.rs

```diff
@@ -287,6 +285,49 @@ fn transform_git_to_http(url: &str) -> String {
     }
 }
 
+fn check_recursiveness(dependency: &Dependency) -> Result<()> {
```

Comment:
This function is not a check (which I would expect to have no side effects). A better name would be `install_subdependencies`